sedaro/src/sedaro/block_client.py

```diff
@@ -112,6 +114,42 @@ def enforce_still_exists(self) -> None:
                 f'The referenced "{self.type}" (id: {self.id}) no longer exists.'
             )
 
+    def clone(self, num: int = 1) -> 'List[BlockClient]':
+        """In a single HTTP request, creates 1+ copies of the Sedaro `Block` corresponding to the `BlockClient` this
+        method is called on.
+
+        Note:
+        - if there is a name attribute on the `Block`, the names of the created `Block`s will have `'clone'` and 32
+        random numbers appended to the end to ensure uniqueness. If the resulting name becomes too long for the `name`
+        field, the clone will fail, in which case you can try again after shortening the name of the original `Block`.
+        Keep in mind that these long names may appear strange when viewed in some parts of the Sedaro UI.
+        - this will not work with `Block`s if the resulting clones violate other unique constraints outside of `name`.
+
+        Args:
+            num (int, optional): the number of copies to make
+
+        Returns:
+            List[BlockClient]: a list of `BlockClients` associated with the created Sedaro `Block`s
+        """
+        new_block = copy.deepcopy(self.data)
+        del new_block[ID]
+
+        if 'name' not in new_block:
+            blocks = [new_block for _ in range(num)]
+
+        else:
+            blocks = []
+            for _ in range(num):
+                b = copy.deepcopy(new_block)
+                b['name'] = f'{b["name"]} - clone {str(uuid.uuid4()).replace("-", "")}'
```

Comment:
This will likely trip out block name validation which restricts names to <= 32 characters. I'm not sure there is much of a case for making identical blocks either.  They will always be different in ways greater than just the name.

Reply:
In other words - I like the clone functionality but we don't need to support cloning more than 1 (in my opinion)

Reply:
Appreciate the feedback! I changed it to just create a single clone and add `clone` on the end.

This is irrelevant now, but just FYI, the max length was actually changed to 100.

![Screenshot 2023-04-12 at 3 26 49 PM](https://user-images.githubusercontent.com/71565231/231563902-bf78f7ef-8d6d-424e-8d4c-253dbf5abafe.png)